src/aap_eda/services/validation.py

```diff
@@ -0,0 +1,97 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import urllib.parse
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+from aap_eda.core import models
+from aap_eda.services.ruleset.activate_rulesets import ACTIVATION_PATH
+
+
+def validate_activation(
+    decision_environment_id: int,
+    rulebook_id: int,
+    user_id: int,
+    extra_var_id: int,
+) -> None:
+    _validate_awx_token(user_id)
+    _validate_decision_environment(decision_environment_id)
+    _validate_extra_var(extra_var_id)
+    _validate_rulebook(rulebook_id)
+    _validate_websocket()
+
+
+def _validate_awx_token(user_id) -> None:
+    tokens = models.AwxToken.objects.filter(user_id=user_id).count()
+    if tokens == 0:
+        raise ValidationError(_("No controller token specified"))
+    elif tokens > 1:
+        raise ValidationError(
+            _(
+                "More than one controller token found, "
+                "currently only 1 token is supported"
+            )
+        )
+
+
+def _validate_websocket() -> None:
+    ws_url = f"{settings.WEBSOCKET_BASE_URL}{ACTIVATION_PATH}"
+    parsed_url = urllib.parse.urlparse(ws_url)
+
+    if parsed_url.scheme not in ["ws", "wss"]:
```

Comment:
@hsong-rh This doesn't seem correct, we should be doing this validation when the app starts up with something like this https://www.dynaconf.com/validation/#validating-in-python-programmatically. If the settings has a bogus value it needs to validated at startup. We shouldn't have to do this with every activation instance, since the settings  doesn't change with each activation.

Reply:
We can get rid of this check and do the validation at startup in a different PR.

Reply:
I proposed to check the encryption key at startup and it was rejected. I think this is the same scenario. So we should not allow any activation if the ws url is wrong. The check is fine but as I said in a previous comment, this is not a validation error. It should follow a different workflow and returns 500, a wrong ws url is a server error, not a client error.

Reply:
@mkanoor @Alex-Izquierdo Changed to keep the websocket checking as it is now, will change it to validate at startup in separate PR